documentation/modules/ROOT/pages/connectors/mongodb.adoc

```diff
@@ -259,6 +225,30 @@ The connector uses the logical name in a number of ways: as the prefix for all t
 You should give each MongoDB connector a unique logical name that meaningfully describes the source MongoDB system.
 We recommend logical names begin with an alphabetic or underscore character, and remaining characters that are alphanumeric or underscore.
 
+// Type: concept
+// Title: How {prodname} MongoDB connectors perform offset consolidation
+// ModuleID: how-debezium-mongodb-connectors-perform-offset-consolidation
+[[mongodb-offset-consolidation]]
+=== Offset consolidation
+The {prodname} MongoDB connector no longer supports `replica_set` connections to sharded MongoDB deployments. 
+As a result, the offsets recorded by connector versions that used the `replica_set` connection mode are incompatible with the current version.
+
+To minimize the way that the connection mode change affects the connector, and to prevent the connector from running unnecessary snapshots, after you upgrade the {prodname} MongoDB connector, it runs an offset consolidation procedure.
+During the consolidation procedure, the connector runs through the following steps to  to reconcile offsets that were recorded by the earlier connector version:
+
+1. Offsets that were recorded by connector versions later than 2.5 are used as-is.
+2. Offsets for events that were captured in `sharded` connection mode from sharded MongoDB deployments, or from MongoDB replica set deployments, are used as-is.
+3. Shard-specific offsets that were recorded by connector versions 2.5.x and earlier are used as-is, if both of the following conditions apply:
+
+* The offsets exist for all current database shards.
+* xref:mongodb-property-allow-offset-invalidation[Offset invalidation] is enabled. +
+If offset invalidation is disabled, the connector fails to start.
+
+
+4.  After the connector completes the preceding steps to process existing offsets, it resumes streaming from the collection, and then commits offsets for new events that it captures. +
```

Comment:
```suggestion
4. After the connector processes existing offsets in the preceding steps, it resumes streaming from the collection, and then commits offsets for new events that it captures. +
```

Reply:
@roldanbob  Good thing you did the pass, I would have missed this. The statement "it resumes streaming from the collection" doesn't really make sense for multiple reasons 

- Changes are technically always streamed from a change stream not from  mongo/db/collection
- Changes stream can be opened against Mongo instance/database/collection
- We actually don't support changes streams scoped to collection yet 

In this case I would use generic "... it resumes streaming changes, and then..."